PROTOCOL.md

```diff
@@ -1046,6 +1048,32 @@ When Row Tracking is enabled (when the table property `delta.enableRowTracking`
   In particular, writers should set `delta.rowTracking.preserved` in the `tags` of the `commitInfo` action to `true` if no rows are updated or copied.
   Writers should set that flag to false otherwise.
 
+# Clustered Table
+
+The Clustered Table feature facilitates the physical clustering of rows that share similar values on a predefined set of clustering columns.
+This enhances query performance when selective filters are applied to these clustering columns through data skipping.
+Clustering columns must be specified during the initial definition of a clustered table, and they can be modified after the table has been created.
+
+A table is defined as a clustered table through the following criteria:
+- When the feature `clustering` exists in the table `protocol`'s `writerFeatures`, then we say that the table is a clustered table.
+  The feature `domainMetadata` is required in the table `protocol`'s `writerFeatures`.
+
+Enablement:
+- The table must be on Writer Version 7.
+- The feature `clustering` must exist in the table `protocol`'s `writerFeatures`.
+
+## Writer Requirements for Clustered Table
+
+When the Clustered Table is supported (when the `writerFeatures` field of a table's `protocol` action contains `clustering`), then:
+- Writers must write out [per-file statistics](#per-file-statistics) and per-column statistics for clustering columns in `add` action. 
+  Failure to collect per-column statistics for clustering columns will result in an error when defining a clustered table or making changes to the clustering columns.
+- Writers must track clustering column names in a `domainMetadata` action with `delta.clustering` as the `domain` and a `configuration` containing all clustering column names.
```

Comment:
Multiple notes on this:
1. Let's put this first. This seems like the starting point.
2. How are the column names stored inside this `domainMetadata`? As a comma separated list? What if the column name contains a comma? (Using physical names avoids the latter issue, I think. Note that the spec does _not_ currently constrain the physical column names, but Parquet doesn't support a particular set of characters in column names, so there can't be commas in physical column names.)

Reply:
> Let's put this first. This seems like the starting point.

Moved to the first bullet.

> How are the column names stored inside this domainMetadata? As a comma separated list? What if the column name contains a comma? (Using physical names avoids the latter issue, I think. Note that the spec does not currently constrain the physical column names, but Parquet doesn't support a particular set of characters in column names, so there can't be commas in physical column names.)

Good point. I think we can use encoding that supports special characters like comma. I am thinking to 
encode column names into a Json array. But let me know if you have any concern on that.

Reply:
The spec definitely needs to include the schema of the new metadata domain. String, array of strings, etc don't parse the same at all. Whatever we choose, an actual example in the spec would help a lot. 

Using the physical column names helps because as you say they won't ever contain whitespace or commas that could confuse a string splitter.

Reply:
Sounds good, I added an example json encoded `DomainMetadata` action at the end of this section below